CMakeLists.txt

```diff
@@ -131,6 +131,8 @@ if (${warnings_as_errors})
         # Make warning as error
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wunused-variable -Wmaybe-uninitialized")
```

Comment:
You might need to differentiate between Clang and GCC. Clang looks like it might not have `-Wmaybe-uninitialized`.

Reply:
..and OSX